src/Common/Scheduler/createResourceManager.cpp

```diff
@@ -44,9 +50,14 @@ class ResourceManagerDispatcher : public IResourceManager
                 if (classifier->has(resource_name))
                     return classifier->get(resource_name);
             }
-            throw Exception(ErrorCodes::RESOURCE_ACCESS_DENIED, "Access denied to resource '{}'", resource_name);
+            if (settings.throw_on_unknown_workload)
+                throw Exception(ErrorCodes::RESOURCE_ACCESS_DENIED, "Access denied to resource '{}'", resource_name);
```

Comment:
Maybe we can make the exception message a bit more clear:
```suggestion
                throw Exception(ErrorCodes::RESOURCE_ACCESS_DENIED, "Could not access resource '{}'. Please check `throw_on_unknown_workload` setting", resource_name);
```